--- processing.js	2009-05-23 00:03:36.000000000 +0100
+++ processing.mine.js	2009-05-22 15:41:08.000000000 +0100
@@ -1808,19 +1808,23 @@
        object[ name ] = fn;
     }
   };
-
+  
+  p.inject = function inject(code){
+    (function(Processing){with (p){
+      eval(parse(code, p));
+    }})(p);
+  }
+  
   p.init = function init(code){
     p.stroke( 0 );
     p.fill( 255 );
   
     // Canvas has trouble rendering single pixel stuff on whole-pixel
     // counts, so we slightly offset it (this is super lame).
-     curContext.translate( 0.5, 0.5 );    
+    curContext.translate( 0.5, 0.5 );    
         
     if ( code ) {      
-      (function(Processing){with (p){        
-        eval(parse(code, p));        
-      }})(p);      
+      p.inject(code);
     }
   
     if (p.setup) {
@@ -1951,4 +1955,4 @@
   return p;
 }
 
-})();
\ No newline at end of file
+})();
